functions/src/aqi-calculation/index.ts

```diff
@@ -161,8 +208,34 @@ exports.calculateAqi = functions.pubsub
       };
 
       currentData[purpleAirId] = currentSensorData;
+
+      // Update the AQI circular buffer for this element
+      const sensorDocRef = firestore.collection('/sensors').doc(docId);
+
+      const status =
+        knownSensor.data().aqiBufferStatus ?? bufferStatus.DoesNotExist;
+      // If the buffer status is In Progress we don't update the buffer at all
+      if (status === bufferStatus.Exists) {
+        // The buffer exists, proceed with normal update
+        let aqiBufferIndex: number = knownSensor.data().aqiBufferIndex;
+        const aqiBuffer: Array<AqiBufferElement> = knownSensor.data().aqiBuffer;
+        aqiBuffer[aqiBufferIndex] = aqiBufferData;
+        /* eslint-disable-next-line no-magic-numbers */
+        aqiBufferIndex = (aqiBufferIndex + 1) % aqiBuffer.length;
+        await sensorDocRef.update({
+          aqiBufferIndex: aqiBufferIndex,
+          aqiBuffer: aqiBuffer,
+        });
+      } else if (status === bufferStatus.DoesNotExist) {
+        // Populate the buffer with default values, skip this sensor
```

Comment:
By "skip this sensor", do you mean we don't add the most recent AQI to the buffer?